### PR TITLE
Customer patience bar

### PR DESCRIPTION
**Pull request by metcalf-putnam**

### CODE REVIEW

File: characters/customers/base_customer/bumper.gd [hunk 1]
```diff
@@ -15,18 +15,17 @@ func _ready() -> void:
 func _calculate_radius() -> void:
 	radius =  $CollisionShape2D.shape.radius
 	
-func set_target(boolean) -> void:
-	is_target = boolean
-	
 func unload_laundry() -> Node2D:
 	var laundry_to_give := .unload_laundry()
+	set_target(false)
```
Comment (metcalf-putnam) [inline characters/customers/base_customer/bumper.gd:20]: set_target is function in base class

File: characters/customers/effects/patience_change/patience_meter.gd [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+	update_bars()
+	
+func update_patience(patience_change : float) -> void:
+	patience += patience_change
+	update_bars()
+	show_change(patience_change)
+	if patience >= max_patience:
+		$HappyParticles.emitting = true
+	else:
+		$HappyParticles.emitting = false
```
Comment (eviedangergalois) [inline characters/customers/effects/patience_change/patience_meter.gd:24]: Why not just $HappyParticles.emitting = patience >= max_patience

File: characters/customers/effects/patience_change/patience_meter.gd [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+	if patience < max_patience * warning_pct:
+		$PatienceParticles.emitting = true
```
Comment (eviedangergalois) [inline characters/customers/effects/patience_change/patience_meter.gd:26]: Is it possible for PatienceParticles to still be emitting if the new patience is greater than max_patience * warning_pct?
Might be safer to just do $PatienceParticles.emitting = patience < max_patience * warning_pct

File: characters/customers/effects/patience_change/patience_meter.gd [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+	if patience <= 0:
+		emit_signal("patience_exhausted")
+	
+func show_change(change : float) -> void:
+	var patience_point = preload("res://characters/customers/effects/patience_change/patience_point.tscn").instance()
+	$PointChange.call_deferred("add_child", patience_point)
+	patience_point.call_deferred("set_and_show", round(change * 100))
+	
+func is_max() -> bool:
+	return true if patience >= max_patience else false
```
Comment (eviedangergalois) [inline characters/customers/effects/patience_change/patience_meter.gd:36]: why not just return patience >= max_patience

**Pull request merged**